### PR TITLE
[PropertyAccess] Fix regression

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? |  |
| Tests pass? | yes |
| Fixed tickets | #18437 |
| License | MIT |
| Doc PR | - |

All credits go to @MisatoTremor 

I don't measure any perf impact.

### CODE REVIEW

File: src/Symfony/Component/PropertyAccess/Tests/Fixtures/TestClass.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyAccess\Tests\Fixtures;
+
+class TestClass
```
Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyAccess/Tests/Fixtures/TestClass.php]: borrowed from 2.7

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php [hunk 2]
```diff
@@ -181,6 +181,7 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
                     if ($propertyPath->isIndex($i)) {
                         if ($overwrite = !isset($zval[self::REF])) {
                             $ref = &$zval[self::REF];
+                            $ref = $zval[self::VALUE];
```
Comment (staabm) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: Looks like the prev line is useless now?

Comment (staabm) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: Or `$ref` is a typo

Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: Not a typo: the prev line turns `$zval[self::REF]` into a reference, which required for writeIndex to work.

**Comment by xabbuh** [comment]
:+1: LGTM

**Comment by stof** [comment]
:+1: 

Status: reviewed

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php (hunk 2 above)
Comment (vsoroka) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: why not change these lines to more readable `$zval[self::REF] = &$zval[self::VALUE]`?

**Pull request merged**